kolibri/core/discovery/models.py

```diff
@@ -10,6 +12,14 @@ class NetworkLocation(models.Model):
     which can be used to sync content or data.
     """
 
+    class Meta:
+        ordering = ["added"]
+
+    # for statically added network locations: `id` will be a random UUID
+    # for dynamically discovered devices: `id` will be the device's `instance_id`
+    id = models.CharField(
+        primary_key=True, max_length=36, default=uuid.uuid4, editable=False
```

Comment:
The ID has been modified so that this table can be used for both static (saved) and dynamic (discovered) network resources.

There are two proxy interfaces to the table for each of these use cases.

The initial reason for keeping one table was to avoid making unnecessary modifications to the channel import wizard front end.  But I kinda like how it turned out.

Reply:
Does any code depend on this being a UUID? Pre-existing `NetworkLocation` objects will have an integer id (which I assume will be turned into a string by the migration), so they would not be UUIDs.

Reply:
It kind of does. it's possible that a single kolibri instance (with the same ID) could be saved several times with different addresses (for example in moving between different network environments, or once in a VPN, once on some specific LAN).  Whereas with the dynamic addresses, we want them to be in a 1:1 relationship with `instance_id`, so this seemed like an okay way to support both in the same table.

Another option would be to introduce a more complex constraint for uniqueness and a more complex query system for dynamic vs. static interfaces. But that seemed unnecessary for now, this does the job as long as nobody starts saving static locations with their `instance_id` for an ID.

Is it too hacky?

Reply:
I should clarify, it's fine if the migrated saved location entries have numeric IDs, they just need to be unique with respect to `instance_id`s. Having instance ID's as a private key is reserved for dynamic entries.

Can you think of a simpler way?

Reply:
FYI, I'm addressing this issue by making a better defined interface for `DynamicNetworkAddress`'s.

Reply:
The migration issue that @rtibbles mentions is a concern here (for existing data), as it could break upgrades if there are existing saved NetworkLocations with numeric IDs.

The need for maintaining uniqueness could be addressed by switching away from adding the DynamicNetworkLocation model, to a more transient cached list of peers -- including a general suggestion about this as a top-level comment.

Reply:
Thanks for this feedback!  The latest push uses `diskcache` instead (for now).